ci-operator/README.md

```diff
@@ -0,0 +1,154 @@
+- `master-sidecar.yaml`: spins up a simple openshift control plane as a sidecar
+  and waits for the `COMMAND` specified to the template to be executed, before
+  itself exiting. The test container is given access to the generated
+  configuration and the `admin.kubeconfig`.
+- `os.yaml`:
+
+
+# Using a template
+
+A template can be used as a stage in `ci-operator` with the `--template`
+argument.  The new template stage is added at runtime to those read from the
+configuration file.  The name of the template, or the name of the file without
+the extension, becomes the name of the stage, so it can be used as an argument
+to `--target`.
+
+When processing the template, `ci-operator` passes several parameters with
+information about the tests (e.g. the job name, which namespace it's operating
+into, URLs for images).  These parameters are described in its usage message.
+
+Logs from pods that fail will be fetched and stored if collection of artifacts
+is enabled (using the `--artifacts-dir` flag).  If part of the output is
+necessary even in case of success, it can be written to the artifact directory.
+
+Below is a sample job (extracted from `cluster/ci/config/prow/config.yaml`)
+with comments added to describe the relevant sections.  For a complete
+description of the fields, see prow's documentation:
+
+https://github.com/kubernetes/test-infra/tree/master/prow
+
+This job uses the `cluster-launch-src.yaml` template to provision a GCP cluster
+and run an e2e test script.  Note that there currently is a lot of unnecessary
+complexity and duplication, which should be cleared in the future.
+
+
+```yaml
+- name: pull-ci-kubernetes-descheduler-e2e-gce-3.10
+  agent: kubernetes
+  always_run: true
+  context: ci/prow/e2e
+  # each branch needs its own ci-operator config
+  branches:
+  - release-3.10
+  rerun_command: "/test e2e"
+  trigger: "((?m)^/test( all| e2e),?(\\s+|$))"
+  decorate: true
+  skip_cloning: true
+  spec:
+    serviceAccountName: ci-operator
+    volumes:
+    # ci-operator template, loaded into a configmap in the ci cluster
+    - name: job-definition
+      configMap:
+        name: prow-job-cluster-launch-src
+    # configuration for gcp clusters, shared by all jobs
+    - name: cluster-profile
+      projected:
+        sources:
+        - secret:
+            name: cluster-secrets-gcp
+        - configMap:
+            name: cluster-profile-gcp
+    # the actual ci-operator container
+    containers:
+    - name: test
+      image: ci-operator:latest
+      volumeMounts:
+      - name: job-definition
+        mountPath: /usr/local/e2e-gcp
+        subPath: cluster-launch-src.yaml
+      - name: cluster-profile
+        mountPath: /usr/local/kubernetes-descheduler-e2e-cluster-profile
+      env:
+      # unique identifier: names gcp instances, etc.
+      - name: JOB_NAME_SAFE
+        value: kubernetes-descheduler-e2e
+      - name: CLUSTER_TYPE
+        value: gcp
+      # ci-operator configuration file
```

Comment:
Is this a reference _to_ the config file, or the complete config file value, or params to inject into a template config file?...